x/pairing/keeper/grpc_query_user_entry.go

```diff
@@ -31,9 +32,24 @@ func (k Keeper) UserEntry(goCtx context.Context, req *types.QueryUserEntryReques
 	// support legacy
 	project, vrfpk_proj, err := k.GetProjectData(ctx, userAddr, req.ChainID, epochStart)
 	if err == nil {
-		allowedCU := project.Policy.EpochCuLimit
+		plan, err := k.subscriptionKeeper.GetPlanFromSubscription(ctx, project.GetSubscription())
+		if err != nil {
+			return nil, err
+		}
+
+		planPolicy := plan.GetPlanPolicy()
+
+		// geolocation is a bitmap. common denominator can be calculated with logical AND
+		geolocation := project.AdminPolicy.GeolocationProfile & project.SubscriptionPolicy.GeolocationProfile & planPolicy.GeolocationProfile
+
+		allowedCU := commontypes.FindMin([]uint64{
```

Comment:
the allowed CU should also consider if the project has below allowedCU  per epoch left.
call project.verifyCU